src/figuresClasses.ts

```diff
@@ -2,10 +1,90 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.a + this.b
+    ) {
+      throw new Error(`Invalid parameters for ${this.shape}`);
+    }
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(`The sides of the ${this.shape} must be greater than 0`);
+    }
+  }
+
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+    const area: number =
+      Math.round(
+        Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+      ) / 100;
+
+    return area;
+  }
+}
+
+export class Circle implements Figure {
+  readonly shape: Shape;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
 
-export class Rectangle implements Figure {}
+    if (this.radius <= 0) {
+      throw new Error(`Radius of ${this.shape} must be greater than 0`);
+    }
+  }
+
+  getArea(): number {
+    const area = Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
```

Comment:
The area calculation for the `Circle` class uses `Math.floor` to round the area to two decimal places. This is inconsistent with the `Triangle` class, which uses `Math.round`. Consider using the same rounding method for consistency across all shape classes.